lib/marked.js

```diff
@@ -1350,7 +1363,7 @@ function splitCells(tableRow, count) {
   }
 
   for (; i < cells.length; i++) {
-    cells[i] = cells[i].replace(/\\\|/g, '|');
+    cells[i] = cells[i].trim().replace(/\\\|/g, '|');
```

Comment:
Worth noting in a comment that leading or trailing whitespace is ignored per the spec.